apis/tempo/v1alpha1/microservices_types.go

```diff
@@ -94,6 +101,118 @@ type MicroservicesList struct {
 	Items           []Microservices `json:"items"`
 }
 
+// ModeType is the authentication/authorization mode in which LokiStack Gateway will be configured.
+//
+// +kubebuilder:validation:Enum=openshift
+type ModeType string
+
+const (
+	// Static mode asserts the Authorization Spec's Roles and RoleBindings
+	// using an in-process OpenPolicyAgent Rego authorizer.
+	OpenShift ModeType = "openshift"
+)
+
+type TenantsSpec struct {
+	// Mode defines the multitenancy mode.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default:=openshift
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:select:static","urn:alm:descriptor:com.tectonic.ui:select:dynamic","urn:alm:descriptor:com.tectonic.ui:select:openshif","urn:alm:descriptor:com.tectonic.ui:select:openshift"},displayName="Mode"
+	Mode ModeType `json:"mode"`
+
+	// Authentication defines the lokistack-gateway component authentication configuration spec per tenant.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authentication"
+	Authentication []AuthenticationSpec `json:"authentication,omitempty"`
+	// Authorization defines the lokistack-gateway component authorization configuration spec per tenant.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authorization"
+	Authorization *AuthorizationSpec `json:"authorization,omitempty"`
+}
+
+// PermissionType is a LokiStack Gateway RBAC permission.
+//
+// +kubebuilder:validation:Enum=read;write
+type PermissionType string
+
+const (
+	// Write gives access to write data to a tenant.
+	Write PermissionType = "write"
+	// Read gives access to read data from a tenant.
+	Read PermissionType = "read"
+)
+
+// SubjectKind is a kind of LokiStack Gateway RBAC subject.
```

Comment:
```suggestion
// SubjectKind is a kind of Microservices Gateway RBAC subject.
```
@pavolloffay I use almost the same structures for the oidc implementation. Do we want to put them in a single PR to avoid conflicts? :smile:

Reply:
yeah we will need to do that

Reply:
ok, i can start working on that tomorrow.